controller/src/main/java/io/pravega/controller/server/eventProcessor/RequestHandler.java

```diff
@@ -19,6 +20,6 @@
  * @param <Request> Type of request this handler will process.
  */
 @FunctionalInterface
-public interface RequestHandler<Request extends ControllerEvent> {
-    CompletableFuture<Void> process(Request request);
+interface RequestHandler<Request extends ControllerEvent> {
+    CompletableFuture<Void> process(Request request, final EventProcessor.Writer<Request> writer);
```

Comment:
The use of this writer isn't consistent, so I'm confused about the motivation to have it in this signature. Could you elaborate, please?

Reply:
Commit request handler needed it, hence added it to the signature for now.. i am cleaning this up with 1738 and letting commit request handler access writer the way autoscalerequest handler uses it.. by using the one in streammetadatatasks. but thats some more refactoring and i decided to take all of that up as part of 1738